### PR TITLE
Add support for FreeIPA deployment

### PR DESCRIPTION
**Pull request by stbenjam**
This deploys a FreeIPA instance, registers the Katello to it, and
configures Realm and an auth source in Katello.  Based on @ehelms PR https://github.com/Katello/katello-deploy/pull/219

One thing it doesn't do is create the Realm AR object, because hammer doesn't support it yet https://github.com/theforeman/hammer-cli-foreman/pull/240

### CODE REVIEW

**Comment by ehelms** [comment]
Mind rebasing now that #219 is merged? I'll take a look at and test this once you do.

File: playbooks/freeipa/tasks/main.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  yum: name=ipa-server-dns state=latest
+
+- name: 'Install Haveged (faster population of PRNG)'
+  yum: name=haveged state=latest
+
+- name: 'Start Haveged'
+  service: name=haveged state=started
+
+- name: 'Fix /etc/hosts record for self'
+  lineinfile: dest=/etc/hosts regexp=".*{{ ansible_nodename }}.*localhost.*" line="{{ freeipa_ip }} {{ ansible_nodename }}" state=present
```
Comment (stbenjam) [inline playbooks/freeipa/tasks/main.yaml]: Newer vegarants create a host record like this:

`127.0.0.1 myhost myhost.example.com localhost.domain localhost`

Which is totally incompatbile with freeipa, it complains about the fqdn resolving to 127.0.0.1. So this line, replaces that with a `192.168.121.2 myhost.example.com myhost` and localhost only is restored below.

**Comment by stbenjam** [comment]
Rebased

**Comment by ehelms** [comment]
Is the setup and configuration of freeipa specific to our setup? I have no issues having our own, but thought I'd check if we should try to make use of a community playbook such as https://galaxy.ansible.com/gregswift/ipaserver/

**Comment by stbenjam** [comment]
To test, vagrant up the box, and it'll install FreeIPA and register your centos7-katello-nightly to it, and configure it for Realm and auth.

To verify auth:

1. Visit https://centos7-freeipa-nightly.example.com/ (you need a hosts record on your local box for this - it doesn't like IP access), you can login with admin/changeme.
2. Create some user for yourself
3. Login again to https://centos7-freeipa-nightly.example.com/ as your new user to change your password
4. This user can now login to Foreman UI, too

To verify realm:
No need to really go through the whole provisioing process of a client, just check Realm feature is on the smart proxy page.

**Comment by stbenjam** [comment]
> Is the setup and configuration of freeipa specific to our setup? I have no issues having our own, but thought I'd check if we should try to make use of a community playbook such as https://galaxy.ansible.com/gregswift/ipaserver/

Yea, most of the playbook is specific to setting up Foreman itself.  I could use that one, but it doesn't buy me much.  We'd need basically all of the playbook as I have it,  and that ipaserver playbook would replace the package install and the 1 shell command to install IPA.

**Comment by ehelms** [comment]
Tested this out and hit:

```
Checking DNS forwarders, please wait ...

stderr: ipa         : ERROR    DNS server 8.8.8.8: query '. SOA': The DNS operation timed out after 10.0000081062 seconds
ipa.ipapython.install.cli.install_tool(Server): ERROR    DNS server 8.8.8.8: query '. SOA': The DNS operation timed out after 10.0000081062 seconds
```

**Comment by stbenjam** [comment]
Oh right, you're in the office which restricts outbound dns, I can probably turn off the DNS feature entirely, it's not really needed for the kind of testing I had in mind

**Comment by ehelms** [comment]
@stbenjam works for me, when you update it I'll re-test

File: playbooks/freeipa.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+- hosts: server freeipa
```
Comment (akofink) [inline playbooks/freeipa.yml]: Why don't we need `become: true` here for setting up `repositories.yml` inside the `common` role?

File: playbooks/freeipa/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,62 @@
+---
```
Comment (akofink) [inline playbooks/freeipa/tasks/main.yaml]: Would it be more idiomatic to add this as a role and include it in `playbooks/freeipa.yml`?

**Comment by stbenjam** [comment]
Closed in favor of https://github.com/Katello/forklift/pull/263

**Pull request closed without merging**